src/lib/debug.ts

```diff
@@ -6,6 +6,22 @@ if (debug.length > 0) {
 
 const debugContains = (key: string) => debug.indexOf(key) !== -1;
 
+/**
+ * Format and print a message to the browser console if its debug key is set.
+ * The first argument is a boolean; no message will be printed if it is false.
+ * Usually the argument should be one of the debugging constants defined in this file.
+ *
+ * @param enabled whether the message should be output.
+ * @param message object passed to console.log for printing, may contain argument placeholders.
+ * @param params additional parameters to insert into message.
+ */
+export function debugLog(enabled: boolean, message: any, ...params: any[]) {
```

Comment:
This is nice. :) I've been wanting to add something like this. I hadn't done it yet because there are cases where there is code that is executed for the log message which might be slow or cause errors with certain edge cases. So this code doesn't get executed with the standard pattern:
```
if (DEBUG_SOMETHING){
 console.log(runSomeCodeThatMightBeUnsafe());
}
```

But with this function that unsafe code will get run whether the debug flag is enabled or not.
```
debugLog(DEBUG_SOMETHING, runSomeCodeThatMightBeUnsafe());
```

A more verbose but safer way to implement this function would be with:
```
debugLog(DEBUG_SOMETHING, (log) => log(runSomeCodeThatMightBeUnsafe()));
```

Or more concise:
```
debugLog(DEBUG_SOMETHING, () => [runSomeCodeThatMightBeUnsafe()]);
```

Reply:
Coincidentally, I just wrote essentially the same function in CODAP.

Reply:
To @scytacki's point, I think it would be sufficient to add a comment to the `debugLog` function indicating that the arguments will be evaluated regardless of the debug setting and so it should not be used for non-trivial computations.

Reply:
I will add a comment.